internal/resources/metal/virtual_circuit/resource.go

```diff
@@ -422,6 +462,42 @@ func resourceMetalVirtualCircuitUpdate(ctx context.Context, d *schema.ResourceDa
 				return diag.Errorf("garbage in tags: %s", ts)
 			}
 		}
+
+		if d.HasChange("subnet") {
+			needsUpdate = true
+			subnet := metalv1.PtrString(d.Get("subnet").(string))
+			ur.VrfVirtualCircuitUpdateInput.Subnet = subnet
+		}
+
+		if d.HasChange("customer_ip") {
+			needsUpdate = true
+			customer_ip := metalv1.PtrString(d.Get("customer_ip").(string))
+			ur.VrfVirtualCircuitUpdateInput.CustomerIp = customer_ip
+		}
+
+		if d.HasChange("metal_ip") {
+			needsUpdate = true
+			metal_ip := metalv1.PtrString(d.Get("metal_ip").(string))
+			ur.VrfVirtualCircuitUpdateInput.MetalIp = metal_ip
+		}
+
+		if d.HasChange("subnet_ipv6") {
+			needsUpdate = true
+			subnet_ipv6 := metalv1.PtrString(d.Get("subnet_ipv6").(string))
+			ur.VrfVirtualCircuitUpdateInput.SubnetIpv6 = subnet_ipv6
+		}
+
+		if d.HasChange("customer_ipv6") {
+			needsUpdate = true
```

Comment:
We should also add a step to the new test that covers this functionality (the new step would pass in a config that is identical to the config used in test step 1, but does not include ipv6 settings)

Reply:
Looks like I found a bug in the API when adding this step. Its tripping a validation error when subnet_ipv6 is sent as empty/nil rather than removing those details. Will work on getting a fix out for that.

Reply:
I see the new test step and it looks to be passing in CI; resolving this thread.